romancal/skycell/match.py

```diff
@@ -26,88 +26,177 @@ class ImageFootprint:
+                wcs.bounding_box = wcs_bbox_from_shape(image_shape)
+
+            vertex_points = wcs.footprint(center=False)
+        else:
+            # constrain maximum number of vertices to image shape
+            vertices = (
+                image_shape[0] - 1 if vertices[0] >= image_shape[0] else vertices[0],
+                image_shape[1] - 1 if vertices[1] >= image_shape[1] else vertices[1],
+            )
+
+            # build a list of pixel indices that represent equally-spaced edge vertices
+            edge_pixel_indices = np.concatenate(
+                [
+                    np.stack(
+                        [
+                            np.repeat(0, repeats=vertices[1] + 1),
+                            np.round(
+                                np.linspace(
+                                    0,
+                                    image_shape[1],
+                                    num=vertices[1] + 1,
+                                    endpoint=False,
+                                )
+                            ),
+                        ],
+                        axis=1,
+                    ),
+                    np.stack(
+                        [
+                            np.round(
+                                np.linspace(
+                                    0,
+                                    image_shape[0],
+                                    num=vertices[0] + 1,
+                                    endpoint=False,
+                                )
+                            ),
+                            np.repeat(image_shape[1], repeats=vertices[0] + 1),
+                        ],
+                        axis=1,
+                    ),
+                    np.stack(
+                        [
+                            np.repeat(image_shape[0] - 1, repeats=vertices[1] + 1),
+                            np.round(
+                                np.linspace(
+                                    image_shape[1],
+                                    0,
+                                    num=vertices[1] + 1,
+                                    endpoint=False,
+                                )
+                            ),
+                        ],
+                        axis=1,
+                    ),
+                    np.stack(
+                        [
+                            np.round(
+                                np.linspace(
+                                    image_shape[0],
+                                    0,
+                                    num=vertices[0] + 1,
+                                    endpoint=False,
+                                )
+                            ),
+                            np.repeat(0, repeats=vertices[0] + 1),
+                        ],
+                        axis=1,
+                    ),
+                ],
+                axis=0,
+            )
```

Comment:
I think I'd suggest making "vertices" a scalar rather than a tuple "how many points do you want per side".  I could imagine someone wanting to specify different numbers for the different sides, but not for Roman, and it makes this code meaningfully more complicated.  Maybe this block would look more like:
np.array([[zeros, xverts], [yverts, xend], [yend, xend - xverts], [yend - yverts, zeros]])
with initial definitions of zeros = np.zeros(vertices), xverts = np.linspace(0, shape[0], vertices, endpoint=False), ...
which might be more legible?

Reply:
done in 34cbca7d0b75cdab95747a8e0cb744d127bfe730

Reply:
For these, do we want the edges to be at (-0.5, shape - 1) or (0, shape)?  I think I might guess the former but we're doing the latter?

And here is how I would recommend this block (untested!):

```suggestion
    zeros = np.zeros(extra_vertices_per_edge + 1)
    xend = shape[0] - zeros
    yend = shape[1] - zeros
    xverts = np.linspace(0, shape[0], num = extra_vertices_per_edge, endpoint=False)
    yverts = np.linspace(0, shape[1], num = extra_vertices_per_edge, endpoint=False)
    edge_indices = np.array([
        np.stack([zeros, yverts], axis=1),
        np.stack([xverts, yend], axis=1),
        np.stack([xend, yend - yverts], axis=1),
        np.stack([xend - xverts, zeros], axis=1)])
```

I think that's the same thing and it is more legible, at least to me?  The key thing is to be able to track that one is moving along the different sides in the right order, and that's easier for me when I don't need to reparse the linspaces & zeros.

I've written this following the existing (0, shape) logic, though as I mentioned it's possible this should all be shifted by a half pixel, either here or when calling the wcs(...).